kvui.py

```diff
@@ -299,6 +300,39 @@ def apply_selection(self, rv, index, is_selected):
         self.selected = is_selected
 
 
+class AutocompleteHintInput(TextInput):
+    min_chars = NumericProperty(3)
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+        self.dropdown = DropDown()
+        self.dropdown.bind(on_select=lambda instance, x: setattr(self, 'text', x))
+        self.bind(on_text_validate=self.on_message)
+
+    def on_message(self, instance):
+        App.get_running_app().commandprocessor("!hint "+instance.text)
+
+    def on_text(self, instance, value):
+        if len(value) >= self.min_chars:
+            self.dropdown.clear_widgets()
+            ctx: context_type = App.get_running_app().ctx
+            item_names = ctx.item_names._game_store[ctx.game].values()
```

Comment:
This introduces a new mypy issue:
> Invalid index type "str | None" for "dict[str, ChainMap[int, str]]"; expected type "str"  [index]

Reply:
This is technically purely a typing  issue, as the underlying class is a defaultdict. Maybe @ThePhar  for comment, as he made the typing.